src/core/util/string_utils.cpp

```diff
@@ -19,6 +18,21 @@
 
 #include <errno.h>
 #include <string.h>
+#include <array>
+#include <map>
+
+const auto scCharacters2Escape = std::array { '"', '\'', '&', '<', '>' };
```

Comment:
do  you really need these two extra arrays? can't you just have the values directly in the constructor of the map?

Reply:
Just noticed my answer was even wrong. Yes I need it for unescaping.

Reply:
if you needed for later is fine, but for unscaping you can iterate directly over the map:

```
for(const auto& [characters2Escape, escapeSequence] : scEscapeMap){
  if(0 == strncmp(runner, escapeSequence, strlen(escapeSequence))){
    toCopy = characters2Escape;
    toMove = strlen(escapeSequence);
    break;
   }
}
```

Reply:
After a second thought and seeing your code I think it is better to not have them for now. Removed them and used your snippet.

Reply:
amazing!

Reply:
thx